vault/resource_consul_secret_backend_role.go

```diff
@@ -257,17 +271,21 @@ func consulSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) error
 		"node_identities":    "node_identities",
 	}
 
-	_, hasLegacyPolicies := data["policies"]
-	if hasLegacyPolicies {
-		params["policies"] = "policies"
-		if _, ok := d.GetOk("consul_policies"); ok {
-			params["policies"] = "consul_policies"
-		}
+	// Check whether Vault will return consul_policies or policies based on its version.
+	hasNewPolicies, _, err := semver.GreaterThanOrEqual(ctx, client, consts.VaultVersion11)
```

Comment:
Maybe rename to `useAPIVer1` or something like that since we we are adapting our response handler based on the Vault version

Reply:
Since Vault version checks were only recently added I would go with `useAPIVer1`, since it is a new epoch.

Reply:
Ah that makes sense, I don't mind changing it 👍